database/sql/src/main/scala/cromwell/database/slick/MetadataSlickDatabase.scala

```diff
@@ -200,16 +200,9 @@ class MetadataSlickDatabase(originalDatabaseConfig: Config)
         summaryPositionFunction =
           metadataEntries => {
             if (metadataEntries.nonEmpty) {
-              DBIO.successful(metadataEntries.map(_.metadataEntryId.get).max)
+              metadataEntries.map(_.metadataEntryId.get).max
             } else {
-              /*
-              Nothing seen within the window of (previousMetadataEntryId, nextMaxMetadataEntryId].
-              Check if there are more entries above nextMaxMetadataEntryId.
-              If yes, then start next time after nextMaxMetadataEntryId, otherwise reuse previousMetadataEntryId.
-               */
-              dataAccess.existsMetadataEntriesGreaterThanMetadataEntryId(nextMaxMetadataEntryId).result map {
-                existsGreater => if (existsGreater) nextMaxMetadataEntryId else previousMaxMetadataEntryId
-              }
+              previousMaxMetadataEntryId
```

Comment:
I definitely had trouble reasoning about this part so I'm glad it's going away

Reply:
Say:
- The summarizer has previously stopped processing at metadata ID 3000
- The summarizer is running with a window size of 1000
- The summarizer's SQL `select` statement ONLY filters for `status` keys (and similar summary keys)
- Some actor floods metadata with a bunch of non-summary keys (in the past this was call caching):

| MetadataID | WorkflowID | MetadataKey | MetadataValue |   |     Window Calculations |
|------------|------------|-------------|---------------|---|------------------------:|
| 3000       | `aaaaaaaa` | status      | Complete      |   | <-- previous maximum id |
| 3001       | `bbbbbbbb` | callCache   | out0001       |   |         window start: 1 |
| 3002       | `bbbbbbbb` | callCache   | out0002       |   |                       2 |
| 3003       | `bbbbbbbb` | callCache   | out0003       |   |                       3 |
| ⋮          | ⋮          | ⋮           | ⋮             |   |                       ⋮ |
| 3999       | `bbbbbbbb` | callCache   | out0999       |   |                     999 |
| 4000       | `bbbbbbbb` | callCache   | out1000       |   |        window end: 1000 |
| 4001       | `bbbbbbbb` | callCache   | out1001       |   |                         |
| 4002       | `cccccccc` | status      | Failed        |   |    outside the window 😢 |

If we _always_ return the `previousMaxMetadataEntryId`, a table with the rows above and a window size == 1000 will never move forward past `previousMaxMetadataEntryId` == 3000, and rows 4002 and beyond will never be processed.